extensions/wikia/UserLogin/js/UserSignupAjaxValidation.js

```diff
@@ -22,111 +25,109 @@
-		if (response.result === 'ok') {
-			this.wikiaForm.clearInputError(paramName);
+		if (msg !== '') {
+			response.msg = msg;
+			response.result = 'error';
 		} else {
-			this.wikiaForm.showInputError(paramName, response.msg);
+			response.result = 'ok';
 		}
+
+		this.validationHandler(inputName, response);
 	};
 
+	/**
+	 * Handle birthday validation as a collection of inputs as opposed to one input.
+	 * @param {Event} e Browser event like select element blur
+	 */
 	UserSignupAjaxValidation.prototype.validateBirthdate = function (e) {
-		var el = $(e.target),
-			paramName = el.attr('name'),
-			params = this.getDefaultParamsForAjax();
+		var $el, inputName, data;
 
-		if (UserSignup.deferred && typeof UserSignup.deferred.reject === 'function') {
-			UserSignup.deferred.reject();
+		if (this.deferred && typeof this.deferred.reject === 'function') {
+			this.deferred.reject();
 		}
 
-		$.extend(params, {
+		$el = $(e.target);
+		inputName = $el.attr('name');
+		data = {
 			field: 'birthdate',
 			birthyear: this.wikiaForm.inputs.birthyear.val(),
 			birthmonth: this.wikiaForm.inputs.birthmonth.val(),
 			birthday: this.wikiaForm.inputs.birthday.val()
-		});
+		};
 
-		UserSignup.deferred = $.post(
-			wgScriptPath + '/wikia.php',
-			params,
-			this.validationHandler.bind(this, paramName)
-		);
+		this.deferred = this.sendRequest(data)
+			.done(this.validationHandler.bind(this, inputName));
 	};
 
 	/**
-	 * @todo User $.nivana instead
-	 * @returns {{controller: string, method: string, format: string}}
+	 * Apply error and success handling via WikiaForm instance
+	 * @param {string} inputName Name of input to handle messaging
+	 * @param {object} response Data for error/success handling
 	 */
-	UserSignupAjaxValidation.prototype.getDefaultParamsForAjax = function () {
-		return {
-			controller: 'UserSignupSpecial',
-			method: 'formValidation',
-			format: 'json'
-		};
-	};
-
-	UserSignupAjaxValidation.prototype.checkFieldsValid = function () {
-		var isValid = true,
-			inputsToValidate = this.inputsToValidate,
-			i;
-
-		for (i = 0; i < inputsToValidate.length; i++) {
-			if (this.checkFieldEmpty(this.wikiaForm.inputs[inputsToValidate[i]]) ||
-				this.wikiaForm.getInputGroup(inputsToValidate[i]).hasClass('error')) {
-				isValid = false;
-				break;
-			}
+	UserSignupAjaxValidation.prototype.validationHandler = function (inputName, response) {
+		if (response.result === 'ok') {
+			this.wikiaForm.clearInputError(inputName);
+		} else {
+			this.wikiaForm.showInputError(inputName, response.msg);
 		}
-
-		return isValid;
 	};
 
-	UserSignupAjaxValidation.prototype.checkFieldEmpty = function (field) {
-		return field && ((field.is('input') && field.val() === '') || (field.is('select') && field.val() === -1));
+	/**
+	 * Shared AJAX helper method
+	 * @param {object} data
+	 * @returns {jQuery} Promise
+	 */
+	UserSignupAjaxValidation.prototype.sendRequest = function (data) {
```

Comment:
How about giving this a slightly more descriptive name? Like `sendValidationRequest` or `validateInputOnServer`? Since we're hitting the same URL every time but just changing the inputs we want to validate.